psi4/driver/procrouting/response/scf_response.py

```diff
@@ -249,3 +254,130 @@ def _print_output(complete_dict, output):
             var_name = prop['name'].upper().replace("IES", "Y")
             _print_matrix(directions, output[i], var_name)
 
+
+def _print_tdscf_header(**options):
+    core.print_out('\n\n         ---------------------------------------------------------\n'
+                   '         {:^57}\n'.format('TDSCF excitation energies') + '         {:^57}\n'.format(
+                       'by Andrew M. James') + '         ---------------------------------------------------------\n')
+
+    core.print_out("\n\n")
+    core.print_out("        " + ("*" * 80) + "\n")
+    core.print_out("        " + "WARNING:".center(80) + "\n")
+    core.print_out("        " + "TDSCF is experimental results may be inaccurate at this point".center(80) + "\n")
+    core.print_out("        " + ("*" * 80) + "\n\n")
+
+    core.print_out("\n  ==> Requested Excitations <==\n\n")
+    state_info = options.pop('states')
+    for nstate, state_sym in state_info:
+        core.print_out("      {} states with {} symmetry\n".format(nstate, state_sym))
+
+    core.print_out("\n  ==> Options <==\n\n")
+    for k, v in options:
+        core.print_out("     {:<10s}:              {}\n".format(k, v))
+
+    core.print_out("\n")
+
+
+def tdscf_excitations(wfn, **kwargs):
+    """Compute excitations from a scf(HF/KS) wavefunction:
+
+    Parameters
+    -----------
+    wfn : :py:class:`psi4.core.Wavefunction`
+       The reference wavefunction
+    states_per_irrep : list (int), {optional}
+       The solver will find this many lowest excitations for each irreducible representation of the computational point group.
+       The default is to find the lowest excitation of each symmetry. If this option is provided it must have the same number of elements
+       as the number of irreducible representations as the computational point group.
+    triplets : str {optional, ``none``, ``only``, ``also``}
+       The default ``none`` will solve for no triplet states, ``only`` will solve for triplet states only, and ``also``
+       will solve for the requested number of states of both triplet and singlet. This option is only valid for restricted references,
+       and is ignored otherwise. The triplet and singlet solutions are found separately so using the ``also`` option will roughly double
+       the computational cost of the calculation.
+    tda :  bool {optional ``False``}
+       If true the Tamm-Dancoff approximation (TDA) will be employed. For HF references this is equivalent to CIS.
+    e_tol : float, {optional, 1.0e-6}
+       The convergence threshold for the excitation energy
+    r_tol : float, {optional, 1.0e-8}
+       The convergence threshold for the norm of the residual vector
+    max_ss_vectors: int {optional}
+       The maximum number of ss vectors that will be stored before a collapse is done.
+    guess : str
```

Comment:
I'm probably thinking about this the wrong way, but given the first order approximation T_ia = f_ia / (E_i - E_a), I'd expect the Fock matrix to enter the default guess also.

Reply:
For HF/KS, f_ia=0!

Reply:
Ah, yes, how could I forget Brillouin's theorem?   Time for ☕️